deploy-service/common/src/main/java/com/pinterest/clusterservice/cm/ClusterManager.java

```diff
@@ -15,21 +15,20 @@
  */
 package com.pinterest.clusterservice.cm;
 
-import com.pinterest.clusterservice.bean.ClusterBean;
-
 import java.util.Collection;
 
 public interface ClusterManager {
+    void createCluster(String clusterName, Object bean) throws Exception;
 
-    void createCluster(String clusterName, ClusterBean bean) throws Exception;
-
-    void updateCluster(String clusterName, ClusterBean bean) throws Exception;
+    void updateCluster(String clusterName, Object bean) throws Exception;
 
-    ClusterBean getCluster(String clusterName) throws Exception;
+    Object getCluster(String clusterName) throws Exception;
```

Comment:
We should not use Object, we can either use a common interface or bean (like you did before), or templatize this interface.